### PR TITLE
port2csproj + target core!

### PR DESCRIPTION
**Pull request by TerribleDev**
* No more stylecop 
  * We'll have to circle back and fix, but it keeps trying to load assemblies not found in the coreclr
* Comment out xsltransform to convert nunit to junit 
  * xsltransforms currently do not exist but will probably comeback when netstandard 2.0 comes back
* Use the dotnet cli to compile and create packages
* Move AssemblyInfo data to csproj files
* Update log4net to target 2.0.8
* Replace System.Timers.Timer with System.Threading.Timer as System.Timers.Timer is not in netstandard
* Replace Thread.Sleep with Task.Delay as Thread.Sleep is not in netstandard 1.5
* Move SumoLogic.Logging.Common to seperate package
  * The dotnet cli, and the msbuild based nuget pack tasks, do not allow you to easily package dependent project dlls. There are workarounds, but do not work with multiTarget builds, and honestly it feels like it should be a seperate package anyway.
* Only console log on the Desktop CLR
* Target NLog 5.0.0-beta7 only on the core clr
  * I kept the desktop on 4.x since NLog 5 is only in beta.
* I did **not** port the tests projects. Mostly because they didn't need to be ported, and this was enough work as it is

 
see #19 and #22

### CODE REVIEW

**Comment by latkin** [comment]
woot, this is on my list to review this week.

**Comment by latkin** [comment]
Getting this when building with CI bat file first time:

```
DONE UPDATING NUGET PACKAGES

BUILDING SOLUTION ...
=====================
Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

C:\src\terribledev-sumologic-net-appenders\SumoLogic.Logging.Log4Net.Tests\SumoLogic.Logging.Log4Net.Tests.csproj(114,5): error : This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\packages\xunit.core.2.0.0\build\portable-net45+win+wpa81+wp80+monotouch+monoandroid+Xamarin.iOS\xunit.core.props.
C:\src\terribledev-sumologic-net-appenders\SumoLogic.Logging.EnterpriseLibrary.Tests\SumoLogic.Logging.EnterpriseLibrary.Tests.csproj(99,5): error : This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\packages\xunit.core.2.0.0\build\portable-net45+win+wpa81+wp80+monotouch+monoandroid+Xamarin.iOS\xunit.core.props.
C:\src\terribledev-sumologic-net-appenders\SumoLogic.Logging.NLog.Tests\SumoLogic.Logging.NLog.Tests.csproj(115,5): error : This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\packages\xunit.core.2.0.0\build\portable-net45+win+wpa81+wp80+monotouch+monoandroid+Xamarin.iOS\xunit.core.props.
C:\src\terribledev-sumologic-net-appenders\SumoLogic.Logging.Common.Tests\SumoLogic.Logging.Common.Tests.csproj(107,5): error : This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\packages\xunit.core.2.0.0\build\portable-net45+win+wpa81+wp80+monotouch+monoandroid+Xamarin.iOS\xunit.core.props.
```

Clears if I explicitly `./.nuget/nuget.exe restore` from repo root, but this should work out of the box.

File: SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs [hunk 1]
```diff
@@ -60,7 +60,7 @@ public class BufferedSumoLogicAppender : AppenderSkeleton
         public BufferedSumoLogicAppender()
             : this(null, null)
         {
-        }
+		}
```
Comment (latkin) [inline SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs]: ❓

File: SumoLogic.Logging.Nuget.bat [hunk 2]
```diff
@@ -1,6 +1,4 @@
 @echo off
-echo "Building Libraries"
-msbuild /p:Configuration=Release SumoLogic.Logging.CI.csproj 
-echo CREATING NUGET PACKAGES ...
-echo ======================================
-msbuild SumoLogic.Logging.CI.csproj /t:MakeNugetPackages /v:m /p:Configuration=Release
+dotnet pack --output %~dp0SumoLogic.Logging.Nuget --configuration Release SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj
```
Comment (latkin) [inline SumoLogic.Logging.Nuget.bat]: I assume these should each output to a different nuget package file, not all to `SumoLogic.Logging.Nuget`?

File: SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs (hunk 1 above)
Comment (TerribleDev) [inline SumoLogic.Logging.Log4Net/BufferedSumoLogicAppender.cs]: sorry I had a change in this file, and then un-did it. I guess I left the whitespace in by accident

File: SumoLogic.Logging.Nuget.bat (hunk 2 above)
Comment (TerribleDev) [inline SumoLogic.Logging.Nuget.bat]: correct, SumoLogic.Logging.Nuget is the output directory. Each package will come out as `${PackageName}.${PackageVersion}.nupkg`

**Comment by TerribleDev** [comment]
its supposed to run dotnet restore before the build on this line here: https://github.com/SumoLogic/sumologic-net-appenders/pull/35/files#diff-6ee9f301d8821dfdceea09f2967f45a1R30

**Comment by TerribleDev** [comment]
@latkin I can't repro the bad restore you previously mentioned. Do you have the newest version of the dotnet cli installed? In this file it should do a restore. Do you see that output when you run this? https://github.com/SumoLogic/sumologic-net-appenders/pull/35/files#diff-6ee9f301d8821dfdceea09f2967f45a1R30

**Comment by TerribleDev** [comment]
nvm I was able to repro it

**Comment by TerribleDev** [comment]
nvm I was able to repro it 568e54f

**Comment by TerribleDev** [comment]
Here is the issue related to stylecop https://github.com/StyleCop/StyleCop/issues/126

**Comment by TerribleDev** [comment]
@latkin I pushed some updates

* fix whitespace (I realized my text editor was set to tabs and not spaces) 17945c5 
* fix restoring of both older style packages and newer ones	568e54f

File: SumoLogic.Logging.Nuget.bat (hunk 2 above)
Comment (latkin) [inline SumoLogic.Logging.Nuget.bat]: :+1: I was confused

**Comment by latkin** [comment]
Awesome! Builds/tests clean, and sanity check on a desktop and .net core app both work great.

Thank you for the hard work to get this all together! I will get a new nuget package published soon.

**Comment by TerribleDev** [comment]
@latkin Holy CRAPPPPPPPPPPPPPPPPPPPPPP

Ok, so when can I expect my sumologic hoodie 😛

**Comment by TerribleDev** [comment]
Just kidding, but thanks this is awesome.

cc @billpratt

**Comment by latkin** [comment]
@TerribleDev let me see what we can do... 😉 

QQ -- nuget yells at me b/c NLog package for .NET Core is still pre-release, but our package is (as it stands) marked stable.

I am thinking I'll push 1.0.0.3 for Common + Log4Net, but push 1.0.0.3-beta1 for NLog so that one is marked pre-release. Is that the right way to do this?

**Comment by TerribleDev** [comment]
@latkin You can just set the version in the csproj to have that suffix. So just make the version say 1.0.0.3-beta1

You can also pass in versions on the command D line

dotnet pack awesome.csproj /p:Version=1.0.0.3-beta1

**Comment by TerribleDev** [comment]
@latkin that sounds like a solid plan btw

**Comment by billpratt** [comment]
Nice work!

**Pull request merged**